feature/google/src/test/java/com/revenuecat/purchases/google/BillingWrapperTest.kt

```diff
@@ -42,7 +61,87 @@ import java.util.concurrent.CountDownLatch
 
 @RunWith(AndroidJUnit4::class)
 @Config(manifest = Config.NONE)
-class BillingWrapperCommonTest : BillingWrapperTestBase() {
+class BillingWrapperTest {
+
+    private var onConnectedCalled: Boolean = false
+    private var mockClientFactory: BillingWrapper.ClientFactory = mockk()
+    private var mockClient: BillingClient = mockk()
+    private var purchasesUpdatedListener: PurchasesUpdatedListener? = null
+    private var billingClientStateListener: BillingClientStateListener? = null
+    private var handler: Handler = mockk()
+    private var mockDeviceCache: DeviceCache = mockk()
+
+    private var mockPurchasesListener: BillingAbstract.PurchasesUpdatedListener = mockk()
+
+    private var capturedAcknowledgeResponseListener = slot<AcknowledgePurchaseResponseListener>()
+    private var capturedAcknowledgePurchaseParams = slot<AcknowledgePurchaseParams>()
+    private var capturedConsumeResponseListener = slot<ConsumeResponseListener>()
+    private var capturedConsumeParams = slot<ConsumeParams>()
+
+    private lateinit var wrapper: BillingWrapper
+
+    private lateinit var mockDetailsList: List<SkuDetails>
+
+    private var storeProducts: List<StoreProduct>? = null
+
+    private val billingClientOKResult = BillingClient.BillingResponseCode.OK.buildResult()
+    private val appUserId = "jerry"
+    private var mockActivity = mockk<Activity>()
+
+    private val subsGoogleProductType = ProductType.SUBS.toGoogleProductType()!!
+    private val inAppGoogleProductType = ProductType.INAPP.toGoogleProductType()!!
+
+    @Before
+    fun setup() {
```

Comment:
I think this was like this before but we should call `clearAllMocks` here. Additionally, we should reset all properties that should be reset with each test like `purchasesUpdatedListener`, `billingClientStateListener` or `storeProducts`.